src/sagas/index.js

```diff
@@ -0,0 +1,29 @@
+import { all, takeEvery, call, put } from 'redux-saga/effects';
```

Comment:
For an implementation like this it's ok to have it all in one file. But as this grow bigger you are going to have to separate into individual files. Also please note that sagas are not only for API fetching, some features like favorite toggling could be done with sagas.